pkg/api/serialization_test.go

```diff
@@ -131,6 +132,22 @@ func roundTripSame(t *testing.T, group testapi.TestGroup, item runtime.Object, e
 	}
 }
 
+func TestSetControllerConversion(t *testing.T) {
+	rs := &extensions.ReplicaSet{}
+	gv := testapi.Extensions
+	fuzzInternalObject(t, gv.InternalGroupVersion(), rs, rand.Int63())
+
+	data, err := runtime.Encode(gv.Codec(), rs)
+	if err != nil {
+		t.Errorf("unexpected encoding error: %v", err)
+	}
+
+	rc := &v1.ReplicationController{}
+	if err := runtime.DecodeInto(testapi.Default.Codec(), data, rc); err != nil {
```

Comment:
@deads2k @lavalamp can you shed some light on this? When decoding to a versioned rc, the converter doesn't seem to pick up my custom conversion, thus it tries to convert field by field ending up in blowing up. At start I thought it was specific to the LabelSelector->map conversion so I moved those into separate custom conversions but now I think it doesn't even pick up my initial custom conversion (RS->RC)
```
--- FAIL: TestSetControllerConversion (0.00s)
panic: reflect: call of reflect.Value.MapKeys on ptr Value [recovered]
	panic: reflect: call of reflect.Value.MapKeys on ptr Value

goroutine 8 [running]:
panic(0x137e080, 0xc820493820)
	/usr/lib/golang/src/runtime/panic.go:481 +0x3e6
testing.tRunner.func1(0xc8200143f0)
	/usr/lib/golang/src/testing/testing.go:467 +0x192
panic(0x137e080, 0xc820493820)
	/usr/lib/golang/src/runtime/panic.go:443 +0x4e9
reflect.flag.mustBe(0x196, 0x15)
	/usr/lib/golang/src/reflect/value.go:202 +0xa6
reflect.Value.MapKeys(0x1509c20, 0xc8204e19b8, 0x196, 0x0, 0x0, 0x0)
	/usr/lib/golang/src/reflect/value.go:1064 +0x53
k8s.io/kubernetes/pkg/conversion.(*Converter).defaultConvert(0xc820178180, 0x1509c20, 0xc8204e19b8, 0x196, 0x1147320, 0xc8204108d8, 0x195, 0xc82000ec80, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/conversion/converter.go:711 +0xbae
k8s.io/kubernetes/pkg/conversion.(*Converter).convert(0xc820178180, 0x1509c20, 0xc8204e19b8, 0x196, 0x1147320, 0xc8204108d8, 0x195, 0xc82000ec80, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/conversion/converter.go:633 +0xc38
k8s.io/kubernetes/pkg/conversion.(*Converter).convertKV(0xc820178180, 0x7ffb9e84c090, 0xc8204937e0, 0x7ffb9e84c090, 0xc820493800, 0xc82000ec80, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/conversion/converter.go:892 +0x712
k8s.io/kubernetes/pkg/conversion.(*Converter).defaultConvert(0xc820178180, 0x1440960, 0xc8204e19b0, 0x199, 0x143ea20, 0xc8204108d0, 0x199, 0xc82000ec80, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/conversion/converter.go:677 +0x1a55
k8s.io/kubernetes/pkg/conversion.(*Converter).convert(0xc820178180, 0x1440960, 0xc8204e19b0, 0x199, 0x143ea20, 0xc8204108d0, 0x199, 0xc82000ec80, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/conversion/converter.go:633 +0xc38
k8s.io/kubernetes/pkg/conversion.(*Converter).convertKV(0xc820178180, 0x7ffb9e84c090, 0xc820493600, 0x7ffb9e84c090, 0xc820493620, 0xc82000ec80, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/conversion/converter.go:892 +0x712
k8s.io/kubernetes/pkg/conversion.(*Converter).defaultConvert(0xc820178180, 0x1460020, 0xc8204e18c0, 0x199, 0x145d6e0, 0xc8204107e0, 0x199, 0xc82000ec80, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/conversion/converter.go:677 +0x1a55
k8s.io/kubernetes/pkg/conversion.(*Converter).convert(0xc820178180, 0x1460020, 0xc8204e18c0, 0x199, 0x145d6e0, 0xc8204107e0, 0x199, 0xc82000ec80, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/conversion/converter.go:633 +0xc38
k8s.io/kubernetes/pkg/conversion.(*Converter).(k8s.io/kubernetes/pkg/conversion.convert)-fm(0x1460020, 0xc8204e18c0, 0x199, 0x145d6e0, 0xc8204107e0, 0x199, 0xc82000ec80, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/conversion/converter.go:527 +0x84
k8s.io/kubernetes/pkg/conversion.(*Converter).doConversion(0xc820178180, 0x152dbc0, 0xc8204e18c0, 0x151e1c0, 0xc8204107e0, 0x8, 0xc820443860, 0xc820451530, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/conversion/converter.go:564 +0x45e
k8s.io/kubernetes/pkg/conversion.(*Converter).Convert(0xc820178180, 0x152dbc0, 0xc8204e18c0, 0x151e1c0, 0xc8204107e0, 0x8, 0xc820443860, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/conversion/converter.go:527 +0x28d
k8s.io/kubernetes/pkg/runtime.(*Scheme).Convert(0xc820170240, 0x152dbc0, 0xc8204e18c0, 0x151e1c0, 0xc8204107e0, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/runtime/scheme.go:458 +0x91b
k8s.io/kubernetes/pkg/runtime/serializer/versioning.(*codec).Decode(0xc82031ebd0, 0xc8200e6d80, 0x46a, 0x46a, 0x0, 0x7ffb9e813920, 0xc8204107e0, 0x0, 0x0, 0x118, ...)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/runtime/serializer/versioning/versioning.go:156 +0x3c8
k8s.io/kubernetes/pkg/runtime.DecodeInto(0x7ffb9e84c038, 0xc82031ebd0, 0xc8200e6d80, 0x46a, 0x46a, 0x7ffb9e813920, 0xc8204107e0, 0x0, 0x0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/runtime/codec.go:60 +0x98
k8s.io/kubernetes/pkg/api_test.TestSetControllerConversion(0xc8200143f0)
	/home/mkargaki/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/api/serialization_test.go:146 +0x48e
```